swarm/storage/memstore.go

```diff
@@ -62,7 +62,12 @@ a hash prefix subtree containing subtrees or one storage entry (but never both)
   (access[] is a binary tree inside the multi-bit leveled hash tree)
 */
 
-func NewMemStore(d *LDBStore, capacity uint) (m *MemStore) {
+func NewMemStore(params *StoreParams, d *LDBStore) (m *MemStore) {
+
+	capacity := params.CacheCapacity
+	if capacity == 0 {
```

Comment:
I worked on this very recently and a `capacity == 0` meant that the MemStore is just disabled, not that we have a defaultCacheCapacity. This is changing the behaviour we have locked with tests. See `swarm/storage/dpa_test.go:125`